cloudcix_primitives/backup_lxd.py

```diff
@@ -107,58 +114,69 @@ def run_host(host, prefix, successful_payloads):
         if ret["channel_code"] != CHANNEL_SUCCESS:
             return False, fmt.channel_error(ret, f"{prefix+2}: " + messages[prefix+2]), fmt.successful_payloads
+        
+        # Check if response indicates backup exists (not a 404 error)
+        response_msg = ret.get('payload_message', '')
+        if 'error_code":404' not in response_msg and 'not found' not in response_msg.lower():
+            # Backup exists in LXD, return success
+            return True, f"1001: {messages[1001]}", fmt.successful_payloads
+        
+        # 3. Create the backup 
+        ret = rcc.run(payload=payloads['create_backup'])
+        if ret["channel_code"] != CHANNEL_SUCCESS:
+            return False, fmt.channel_error(ret, f"{prefix+3}: " + messages[prefix+3]), fmt.successful_payloads
         if 'payload_message' not in ret or not ret['payload_message']:
-            return False, fmt.payload_error(ret, f"{prefix+3}: " + messages[prefix+3]), fmt.successful_payloads
+            return False, fmt.payload_error(ret, f"{prefix+4}: " + messages[prefix+4] + f"Backup creation command was: {payloads['create_backup']}"), fmt.successful_payloads
+        
         
         try:
             response = json.loads(ret['payload_message'])
             operation_url = response.get("operation")
             if not operation_url:
-                return False, fmt.payload_error(ret, f"{prefix+3}: " + messages[prefix+3] + "Invalid response, no operation URL"), fmt.successful_payloads
-        except (json.JSONDecodeError, TypeError):
-            return False, fmt.payload_error(ret, f"{prefix+3}: " + messages[prefix+3] + "Invalid JSON response"), fmt.successful_payloads
+                return False, fmt.payload_error(ret, f"{prefix+4}: " + messages[prefix+4] + f"Invalid response, no operation URL. Response was: {ret['payload_message']}"), fmt.successful_payloads
+        except (json.JSONDecodeError, TypeError) as e:
+            return False, fmt.payload_error(ret, f"{prefix+4}: " + messages[prefix+4] + f"Invalid JSON response: {e}. Raw response: {ret['payload_message']}"), fmt.successful_payloads
         
         fmt.add_successful('create_backup', ret)
         
-        # 3. Wait for backup to complete
-        ret = rcc.run(payload=payloads['wait_backup'](operation_url))
+        # 4. Wait for backup to complete
+        wait_command = payloads['wait_backup'](operation_url)
+        ret = rcc.run(payload=wait_command)
         if ret["channel_code"] != CHANNEL_SUCCESS:
-            return False, fmt.channel_error(ret, f"{prefix+4}: " + messages[prefix+4]), fmt.successful_payloads
+            return False, fmt.channel_error(ret, f"{prefix+5}: " + messages[prefix+5]), fmt.successful_payloads
         
         fmt.add_successful('wait_backup', ret)
         
-        # 4. Export the backup
+        # 5. Export the backup
         ret = rcc.run(payload=payloads['export_backup'])
         if ret["channel_code"] != CHANNEL_SUCCESS:
```

Comment:
How do we know the backup was exported successfully? 

It is ensuring that the `ret["channel_code"]` is `CHANNEL_SUCCESS` - SSH connection was successful but not that the `ret["payload_code"]` is `SUCCESS_CODE` - payload sent did not raise any errors while being executed on host